draft-ietf-quic-tls.md

```diff
@@ -499,9 +499,17 @@ handshake, new data is requested from TLS after providing received data.
 ### Encryption Level Changes
 
 As keys at a given encryption level become available to TLS, TLS indicates to
-QUIC that reading or writing keys at that encryption level are available.  These
-events are not asynchronous; they always occur immediately after TLS is provided
-with new handshake bytes, or after TLS produces handshake bytes.
+QUIC that reading or writing keys at that encryption level are available.
+
+The events that cause new keys to be available are not asynchronous; they
+always occur immediately after TLS is provided with inputs. Inputs are either
+new handshake bytes or new instructions. The two instructions that this
+document relies upon are the initial signal to start the handshake and - if the
+TLS implementation allows asynchronous certificate validation - an indication
```

Comment:
This seems to assume that certificate validation is done "outside" of the TLS library if it's done asynchronously. I can imagine a design where certificate validation happens "inside" the TLS stack, where the QUIC stack still decides to not block on the result.